torch/csrc/api/src/nn/modules/loss.cpp

```diff
@@ -180,6 +180,33 @@ Tensor TripletMarginLossImpl::forward(
 
 // ============================================================================
 
+TripletMarginWithDistanceLossImpl::TripletMarginWithDistanceLossImpl(
+    const TripletMarginWithDistanceLossOptions& options_) // NOLINT(modernize-pass-by-value)
```

Comment:
Will complier complain if removing `NOLINT(...)`?
I kinda remember c10:optional doesn't work well with std::move. Did you try std::move here?

Reply:
clint complains about this otherwise; I copied the `NOLINT()` from elsewhere in the file.

What would `std::move` look like in this case?  Looks like everything else in this file uses the same format, minus `NOLINT()` if it's not applicable.

Reply:
You can do the following to avoid that lint error.
`
TripletMarginWithDistanceLossImpl::TripletMarginWithDistanceLossImpl(
    TripletMarginWithDistanceLossOptions options_) : 
    options(std::move(options_)) {}
`
But I kinda remember std::move is not working well with c10::optional which is inside options. But if you can try, that would be great.

Reply:
std::move couldn't build once for me locally, then it started building -- not sure if it's a flaky failure.  I've pushed this and if the CI fails, then I'll go back to pass-by-reference